pydra/engine/helpers.py

```diff
@@ -158,13 +159,46 @@ def make_klass(spec):
     finally:
         # wait for the process to exit
         rc = await process.wait()
-    return rc, stdout, stderr
+    if strip:
+        return rc, stdout.rstrip(), stderr
+    else:
+        return rc, stdout, stderr
+
+
+def read_and_display(*cmd, strip=False, hide_display=False):
+    """Capture cmd's stdout
 
+    """
+    try:
+        process = sp.run(cmd, stdout=sp.PIPE, stderr=sp.PIPE)
+    except Exception:
+        # TODO editing some tracing?
+        raise
 
-# run the event loop
-def execute(cmd):
+    if strip:
+        return (
+            process.returncode,
+            process.stdout.decode("utf-8").rstrip(),
+            process.stderr.decode("utf-8"),
+        )
+    else:
+        return (
+            process.returncode,
+            process.stdout.decode("utf-8"),
+            process.stderr.decode("utf-8"),
+        )
+
+
+# run the event loop with coroutine read_and_display_async
+# or run read_and_display if a loop is already running
+def execute(cmd, strip=False):
     loop = get_open_loop()
-    rc, stdout, stderr = loop.run_until_complete(read_and_display(*cmd))
+    if loop.is_running():
+        rc, stdout, stderr = read_and_display(*cmd, strip=strip)
```

Comment:
is it not possible to just await here?

Reply:
not without a refactor to make `Task._run_task()` async.